test/engine-luatest/gh_9548_fixed_size_numeric_types_test.lua

```diff
@@ -320,3 +320,59 @@ g.test_format_change = function(cg)
         end
     end)
 end
+
+-- Check that fixed-size numeric types can be indexed.
+local function test_indexed(cg, params)
+    cg.server:exec(function(params)
+        local ffi = require('ffi')
+        local is_nullable = params.is_nullable
+        local s = box.space.test
+        s:format({{name = 'pk',  type = 'unsigned', is_nullable = is_nullable},
+                  {name = 'i8',  type = 'int8',     is_nullable = is_nullable},
+                  {name = 'u8',  type = 'uint8',    is_nullable = is_nullable},
+                  {name = 'i16', type = 'int16',    is_nullable = is_nullable},
+                  {name = 'u16', type = 'uint16',   is_nullable = is_nullable},
+                  {name = 'i32', type = 'int32',    is_nullable = is_nullable},
+                  {name = 'u32', type = 'uint32',   is_nullable = is_nullable},
+                  {name = 'i64', type = 'int64',    is_nullable = is_nullable},
+                  {name = 'u64', type = 'uint64',   is_nullable = is_nullable},
+                  {name = 'f32', type = 'float32',  is_nullable = is_nullable},
+                  {name = 'f64', type = 'float64',  is_nullable = is_nullable}})
+        s:create_index('i8',  {parts = {{field = 'i8'}}})
```

Comment:
I think it's also worth checking that the hash index works as well.

Reply:
Done.